packages/discord.js/src/util/Components.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+/**
+ * @typedef {Object} BaseComponentData
+ * @property {ComponentType} type
+ */
+
+/**
+ * @typedef {BaseComponentData} ActionRowData
+ * @property {ComponentData[]} components
+ */
+
+/**
+ * @typedef {BaseComponentData} ButtonComponentData
+ * @property {ButtonStyle} style
+ * @property {boolean?} disabled
```

Comment:
```suggestion
 * @property {?boolean} disabled
```
The `?` shouldn't you go first?